src/node_file.cc

```diff
@@ -1226,8 +1240,13 @@ static void Rename(const FunctionCallbackInfo<Value>& args) {
 
   BufferValue old_path(isolate, args[0]);
   CHECK_NOT_NULL(*old_path);
+  THROW_IF_INSUFFICIENT_PERMISSIONS(
+      env, policy::Permission::kFileSystemOut, *old_path);
```

Comment:
Good catch! It actually needs to be both. Fixed in: 706ee25980

Reply:
Why both on the old path? Wouldn't it be In for reading from the old path and Out for writing the new one?

Reply:
It will override the old_path as well. For instance, you deny `out` access to the `.gitignore` file. The following code will work fine if no check in the old_path exists

```js
fs.readFileSync('.gitignore') // throw error
fs.rename('.gitignore', '.gitignore-allowed') // ok
fs.readFileSync('.gitignore-allowed') // ok
```
This behavior would be the same as `fs.rm('.gitignore')`

Reply:
In theory, if you deny `out` access to a specific resource. The process shouldn't be able to write/remove/move it

Reply:
Ah, good catch. Thanks for the explanation. 🙂